modules/javafx.controls/src/main/java/javafx/scene/control/skin/TreeTableRowSkin.java

```diff
@@ -98,46 +96,45 @@ public TreeTableRowSkin(TreeTableRow<T> control) {
 
         updateTreeItem();
 
-        ListenerHelper lh = ListenerHelper.get(this);
-
-        lh.addChangeListener(control.indexProperty(), (ev) -> {
+        control.indexProperty().addListener(new WeakInvalidationListener((x) -> {
             updateCells = true;
-        });
+        }));
```

Comment:
I don't think this needed changing, the index property is directly available on `TreeTableRow` and so shouldn't cause GC issues.